unsafe/src/main/java/org/apache/spark/unsafe/memory/TaskMemoryManager.java

```diff
@@ -58,8 +58,8 @@
   /** The number of entries in the page table. */
   private static final int PAGE_TABLE_SIZE = 1 << PAGE_NUMBER_BITS;
 
-  /** Maximum supported data page size */
-  private static final long MAXIMUM_PAGE_SIZE = (1L << OFFSET_BITS);
+  /** Maximum supported data page size (in bytes). */
+  public static final long MAXIMUM_PAGE_SIZE_BYTES = (1L << OFFSET_BITS);
```

Comment:
I noticed that there's actually a bug in the existing code w.r.t on-heap vs. off-heap mode: in on-heap mode, the maximum page size is bounded by both the number of addressing bits _and_ the size of a long array, so this theoretical upper limit only applies to off-heap mode.